stdlib/src/utils/string_slice.mojo

```diff
@@ -116,6 +116,15 @@ struct StringSlice[
     fn __str__(self) -> String:
         return String(str_slice=self)
 
+    fn __len__(self) -> Int:
+        """Returns the StringSlice byte length.
```

Comment:
Same as above, for consistency with Python (and for `StringSlice` to be consistent with `String`) I think the `__len__` for a string should be its length in Unicode code points.

Similarly, we can promote / add a `byte_length()` method for use in cases where the encoded length is needed.